src/index.ts

```diff
@@ -213,36 +221,60 @@ export class Driver {
         return { response,  messages, index: messageIdx, status: 'ready' }
       }
       case 'input_wait': {
-        const toolUseCount = stage.toolCallIndex!
-        const submessageIdx = stage.submessageIdx!
-        const inputMessage = messages[index-1]
         const newMessage = messages[index]
-
-        this.#logger.info({ m:"message index", index, len: messages.length })
-
-        // when status == 'input_wait' it is always a tool call,
-        // newMessage.content is always a ContentBlockParam[]
-        const submessage = inputMessage.content[submessageIdx] as ToolUseBlock
         const content = newMessage.content as ContentBlockParam[]
-        content.push(await this.call(submessage))
+        const { tool, submessageIdx, toolCallIndex, toolCallLength, toolCallId } =
+          this.parseNextToolCall(stage)
+
+        content.push(await this.call(tool, toolCallId!))
```

Comment:
:three: so we can pass it directly here. But I don't see any other changes to the logic?